### PR TITLE
Minor improvements to GPU IPC

### PR DESCRIPTION
**Pull request by pavanbalaji**
## Pull Request Description

Minor improvements to GPU IPC, including a bug fix.

## Author Checklist
* [x] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [x] Remove xfail from the test suite when fixing a test
* [x] Commits are self-contained and do not do two things at once
* [x] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [x] Passes whitespace checkers
* [x] Passes warning tests
* [x] Passes all tests
* [x] Add comments such that someone without knowledge of the code could understand
* [x] You or your company has a signed contributor's agreement on file with Argonne
* [x] For non-Argonne authors, request an explicit comment from your companies PR approval manager

### CODE REVIEW

**Comment by pavanbalaji** [comment]
test:mpich/ch4/most

**Comment by pavanbalaji** [comment]
test:mpich/ch4/most

**Comment by pavanbalaji** [comment]
test:mpich/warnings/gpu

File: src/mpid/ch4/shm/ipc/src/ipc_send.h [hunk 1]
```diff
@@ -33,7 +33,10 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_IPC_mpi_isend(const void *buf, MPI_Aint count
 
     if (rank != comm->rank && attr.ipc_type != MPIDI_IPCI_TYPE__NONE &&
         data_sz >= attr.threshold.send_lmt_sz) {
-        if (dt_contig) {
+        MPIR_Datatype *dtp;
+        MPIR_Datatype_get_ptr(datatype, dtp);
+
+        if (dt_contig && dtp->true_lb == 0) {
             mpi_errno = MPIDI_IPCI_send_contig_lmt(buf, count, datatype, data_sz, rank, tag, comm,
```
Comment (minsii) [inline src/mpid/ch4/shm/ipc/src/ipc_send.h:40]: I do not see why we cannot support `true_lb>0` as long as it is contiguous. Can we give `sbuf+true_lb` to the receiver (i.e., using `sbuf+true_lb` to get the memory handle) ? I think this was how xpmem originally handled it. But looks like the code was lost when we refactored it to memory handle.

Comment (pavanbalaji) [reply src/mpid/ch4/shm/ipc/src/ipc_send.h:40]: Can you add the code for it?  I'm tied up with other things at the moment.

Comment (pavanbalaji) [reply src/mpid/ch4/shm/ipc/src/ipc_send.h:40]: Note that the query for GPU buffer, and the IPC handle creation need to be modified too.

Comment (minsii) [reply src/mpid/ch4/shm/ipc/src/ipc_send.h:40]: Will do.

Comment (minsii) [reply src/mpid/ch4/shm/ipc/src/ipc_send.h:40]: Emm, looking at the code again. `true_lb` is already added into mem handle. It should work correctly. Is there any test fails with this code?

```
    MPIDI_IPCI_mem_attr_t attr;
    MPIDI_IPCI_get_mem_attr((char *) buf + true_lb, &attr);

    if (rank != comm->rank && attr.ipc_type != MPIDI_IPCI_TYPE__NONE &&
```

Comment (pavanbalaji) [reply src/mpid/ch4/shm/ipc/src/ipc_send.h:40]: Yes, if you try Ken's PR, a bunch of tests are failing.

**Comment by pavanbalaji** [comment]
test:mpich/warnings/gpu
test:mpich/ch4/most

**Comment by pavanbalaji** [comment]
This PR now contains just the minor naming change patch.  A variant of the remaining patches has already gone into master.

test:mpich/warnings/gpu
test:mpich/ch4/most

**Comment by pavanbalaji** [comment]
Tests are clean.  Rebased to master.

**Pull request merged**